### DISCUSSION TITLE
Index lost on restart

### DISCUSSION THREAD

**Discussion by krakky4** [photoprism/photoprism#3743] [Q&A]
Hello there,
I am new to Photoprism but pretty impressed by the UI and features so far.
Running it on an Asustor NAS.
My photo folder is on a different shared folder on the same NAS and mounted as the originals folder via portainer. It is found and photos are indexed,
however

When re-starting the NAS, all indexing is lost and there are no photos displayed unless I start indexing all over.
If I stop indexing, then log off and on again, everything is fine. When stopping the containers in Portainer and re-starting them, everything is fine as well. It is just the NAS-reboot that kills all indexing...

Is this a known issue or something that reminds anybody of possible problem sources? Where can I check?
Thanks in advance
Krakky

Basic information:
(a) PhotoPrism Architecture & Build Number: 
Build 230719-73fa7bbe8

(b) Database Type & Version: 
MariaDB, version?

(c) Operating System Types & Versions: 
Asustor ADM 4.2.4.RL82

**Comment by lastzero** [comment]
It seems you didn't configure permanent storage for the database?

**Reply by krakky4** [reply]
Thankks for replyiing. How would I do that?

**Reply by graciousgrey** [reply]
If you are using Portainer, you can follow this guide: https://docs.photoprism.app/getting-started/portainer/#photoprismstorage

**Reply by lastzero** [reply]
It is important that all folders are located on a persistent volume on your device - ideally specified with an absolute path - to avoid potential data loss after a restart, e.g. if the default application folder managed by Portainer changes or gets lost for a reason beyond our control.

**Reply by krakky4** [reply]
I can not find a problem there. The stack reads
`      - "./storage:/photoprism/storage"     # *Writable* storage folder for cache, database, and sidecar files (DO NOT REMOVE)`
which is the default - I did not change anything for the storage.

Then in the environment variables there is
`PHOTOPRISM_STORAGE_PATH` `/photoprism/storage`

And in volumes the settings are
`/data/compose/4/storage` as Host/volume for the path in container `/photoprism/storage` 

Both, originals folder and storage folder are on the same persistent volume (four RAID HDDs in the NAS).

Anything to work from here?
Thanks!

**Reply by lastzero** [reply]
See the warning and explanation in our Portainer Guide: ". /..." is a path relative to the default application directory that Portakner manages. I've only set up Portakner once to write this guide and could not test it with third-party integrations, such as those provided by NAS vendors. There is no hard guarantee the default application directory isn't temporary for any reason, including bugs in Portainer, bugs in the NAS integration and accidental misconfiguration. As a result, you should not store data there that's important to you and that you want to keep for the next couple of years. Unfortunately, absolute paths always depend on your local environment eg what disks you have. So we can't tell you exactly what to use.

**Reply by krakky4** [reply]
I changed the  `- "./storage:/photoprism/storage"     # *Writable* storage folder for cache, database, and sidecar files (DO NOT REMOVE)`  to an absolute path `- "/volume1/Media/Photoprism/storage:/photoprism/storage"     # *Writable* storage folder for cache, database, and sidecar files (DO NOT REMOVE)` 
The storage folder structure is created there as expected but it is still the same problem: on re-start of the NAS, all indexing is lost.

Photoprism really looked good and suited my needs but it is just not working for me after many hours of trying. So I guess I'll have to find another application :(
Thank you anyways for trying to help.

**Reply by lastzero** [reply]
You probably have the same issue for the "./database" folder of the mariadb service then (same file further down)?
